TrailBlaze/View Controllers/MateDetailViewController.m

```diff
@@ -18,31 +20,87 @@ @interface MateDetailViewController ()
-            [tabBarController setSelectedViewController: navController];
-            sceneDelegate.window.rootViewController = tabBarController;
-            NSLog(@"got run");
+    //start activity indicator animation
+    [InterceptRequest uploadRequest:PFUser.currentUser.objectId receiverID:_thisUser.objectId withCompletion:^(BOOL succeeded, NSError * _Nullable error) {
+        if (succeeded) {
+            NSLog(@"sent intercept request");
             
         } else {
-            NSLog(@"did not get run");
+            NSLog(@"failed to send intercept request");
+        }
+    }];
+     
+}
+
+- (void) parseLiveQuerySetUp {
+    NSString *path = [[NSBundle mainBundle] pathForResource: @"keys" ofType: @"plist"];
+    NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: path];
+
+    NSString *appID = [dict objectForKey: @"appID"];
+    NSString *clientKey = [dict objectForKey: @"clientKey"];
+    liveQueryClient = [[PFLiveQueryClient alloc] initWithServer:@"https://tblaze.b4a.io" applicationId:appID clientKey:clientKey];
+    approvalQuery = [PFQuery queryWithClassName:@"InterceptRequest"];
+//    [approvalQuery whereKey:@"approved" equalTo: [NSNumber numberWithBool:YES]];
+    [approvalQuery whereKey:@"requester" equalTo: PFUser.currentUser.objectId];
+    liveQuerySubscription = [liveQueryClient subscribeToQuery:approvalQuery];
+    __weak typeof(self) weakself = self;
+    [liveQuerySubscription addUpdateHandler:^(PFQuery<PFObject *> * _Nonnull query, PFObject * _Nonnull object) {
+        __strong typeof(self) strongself = weakself;
+        ///delete interceptRequest after approval
+        [object deleteInBackground];
+        if ([object[@"approved"] boolValue] == YES) {
+            [Run retreiveRunPolyline:strongself->_thisUser completion:^(MKPolyline * _Nonnull polyline, NSError * _Nullable err) {
+               if (polyline) {
+                   SceneDelegate *sceneDelegate = (SceneDelegate *)UIApplication.sharedApplication.connectedScenes.allObjects.firstObject.delegate;
+                   UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+                   UITabBarController *tabBarController = [storyboard instantiateViewControllerWithIdentifier:@"TabBarController"];
+                   UINavigationController *navController = tabBarController.viewControllers[1];
+                   HomeViewController *hvc = navController.childViewControllers[0];
+                   hvc.cloudPolyline = polyline;
+                   hvc.cloudUser = strongself->_thisUser;
+                   [tabBarController setSelectedViewController: navController];
+                   sceneDelegate.window.rootViewController = tabBarController;
+                   NSLog(@"got run");
+                   //stop activity indicator animation
+               } else {
+                   NSLog(@"did not get run");
+               }
+           }];
+        } else if ([object[@"approved"] boolValue] == NO){
```

Comment:
`[object[@"approved"] boolValue]`

maybe pull out the derivation of this value and reuse it so we don't have an `if`/`else if` without an `else`